gcloud-java-compute/src/main/java/com/google/gcloud/compute/Operation.java

```diff
@@ -0,0 +1,793 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Google Compute Engine operations. Operation identity can be obtained via {@link #operationId()}.
+ * {@link #operationId()} returns {@link GlobalOperationId} for global operations,
+ * {@link RegionOperationId} for region operations, and {@link ZoneOperationId} for zone operations.
+ * To get an {@code Operation} object with the most recent information, use
+ * {@link #reload(Compute.OperationOption...)}.
+ */
+public final class Operation implements Serializable {
```

Comment:
Per the issue we had with `Storage`. if our API returns a type X we should provide a way for creating an instance of that type (when applicable) or not have it as final (so it will be easier to mock it). We should still make the constructor package scope (and make some of the methods final - such as hashCode/equals).